sdk/monitor/monitor-query/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,389 @@
+# Troubleshooting Azure Monitor Query client library issues
+
+This troubleshooting guide contains instructions to diagnose frequently encountered issues while using the Azure Monitor Query client library for JavaScript.
+
+## Table of contents
+
+- [General troubleshooting](#general-troubleshooting)
+  - [Enable client logging](#enable-client-logging)
+  - [Troubleshooting authentication issues with logs and metrics query requests](#authentication-errors)
+- [Troubleshooting logs query](#troubleshooting-logs-query)
+  - [Troubleshooting insufficient access error](#troubleshooting-insufficient-access-error-for-logs-query)
+  - [Troubleshooting invalid Kusto query](#troubleshooting-invalid-kusto-query)
+  - [Troubleshooting empty log query results](#troubleshooting-empty-log-query-results)
+  - [Troubleshooting server timeouts when executing logs query request](#troubleshooting-server-timeouts-when-executing-logs-query-request)
+  - [Troubleshooting partially successful logs query requests](#troubleshooting-partially-successful-logs-query-requests)
+- [Troubleshooting metrics query](#troubleshooting-metrics-query)
+  - [Troubleshooting authorization failed error](#troubleshooting-authorization-failed-error-for-metrics-query)
+  - [Troubleshooting unsupported granularity](#troubleshooting-unsupported-granularity-for-metrics-query)
+
+## General troubleshooting
+
+### Enable client logging
+
+To troubleshoot issues with Azure Monitor Query library, it's important to first enable logging to monitor the behavior of the application. The errors and warnings in the logs generally provide useful insights into what went wrong and sometimes include corrective actions to fix issues.
```

Comment:
Maybe we should just link back to the logging section in the README or will that be replaced with this? https://github.com/azure/azure-sdk-for-js/tree/main/sdk/monitor/monitor-query#logging

Reply:
I feel like this is generic enough that we can just link to https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/logger/README.md or something instead of repeating it here what do you think?

Reply:
I was thinking the same initially. But other languages (Python and Java) added [code snippets](https://github.com/rakshith91/azure-sdk-for-python/blob/dc61bb5382e9309687bc54aed4d2101951092514/sdk/monitor/azure-monitor-query/TROUBLESHOOTING.md#enable-client-logging). Due to this, I added it too, just to be consistent. 

Should we remove the code snippets from all languages and just point to the logger documentation? @scottaddie